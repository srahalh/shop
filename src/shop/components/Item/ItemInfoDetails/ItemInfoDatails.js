import React from "react";
import PropTypes from "prop-types";

import "./ItemInfoDetails.scss";

const ItemInfoDatails = ({
  model,
  brand,
  price,
  cpu,
  ram,
  so,
  displayResolution,
  battery,
  dimentions,
  primaryCamera,
  secondaryCmera: secondaryCamera,
  weight,
}) => {
  return (
    <section className="ItemInfoDetails__container">
      {model && <h3 className="ItemInfoDetails__title">{model}</h3>}
      {brand && (
        <p className="ItemInfoDetails__attribute">
          <strong>Brand: </strong> {brand}
        </p>
      )}
      {cpu && (
        <p className="ItemInfoDetails__attribute">
          <strong>CPU: </strong> {cpu}
        </p>
      )}
      {ram && (
        <p className="ItemInfoDetails__attribute">
          <strong>RAM: </strong>
          {ram}
        </p>
      )}
      {so && (
        <p className="ItemInfoDetails__attribute">
          <strong>SO: </strong>
          {so}
        </p>
      )}
      {displayResolution && (
        <p className="ItemInfoDetails__attribute">
          <strong>Display Resolution: </strong>
          {displayResolution}
        </p>
      )}
      {battery && (
        <p className="ItemInfoDetails__attribute">
          <strong>Battery: </strong>
          {battery}
        </p>
      )}
      {dimentions && (
        <p className="ItemInfoDetails__attribute">
          <strong>Dimentions: </strong>
          {dimentions}
        </p>
      )}
      {primaryCamera && (
        <p className="ItemInfoDetails__attribute">
          <strong>Primary Camera: </strong>
          {primaryCamera}
        </p>
      )}
      {secondaryCamera && (
        <p className="ItemInfoDetails__attribute">
          <strong>Secondary Camera: </strong>
          {secondaryCamera}
        </p>
      )}
      {weight && (
        <p className="ItemInfoDetails__attribute">
          <strong>Weight: </strong>
          {weight}
        </p>
      )}
      {price && (
        <p className="ItemInfoDetails__price">
          Price: <strong>{price} â‚¬</strong>
        </p>
      )}
    </section>
  );
};

ItemInfoDatails.propTypes = {
  model: PropTypes.string,
  brand: PropTypes.string,
  price: PropTypes.string,
  cpu: PropTypes.string,
  ram: PropTypes.string,
  so: PropTypes.string,
  displayResolution: PropTypes.string,
  battery: PropTypes.string,
  dimentions: PropTypes.string,
  primaryCamera: PropTypes.string,
  secondaryCmera: PropTypes.string,
  weight: PropTypes.string,
};

export default ItemInfoDatails;
