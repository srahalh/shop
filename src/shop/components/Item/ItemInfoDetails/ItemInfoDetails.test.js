import React from "react";
import { render } from "@testing-library/react";
import ItemInfoDatails from "./ItemInfoDatails";
const products = require("../../../test/products.json");

describe("ItemInfoDatails component", () => {
  test("Should render ItemInfoDatails Component", () => {
    const props = products[0]
    const { asFragment  } = render(<ItemInfoDatails {...props} /> );
    expect(asFragment ()).toMatchSnapshot();
  });
});
