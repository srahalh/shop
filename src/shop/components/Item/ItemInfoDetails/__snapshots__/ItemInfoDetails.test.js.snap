// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemInfoDatails component Should render ItemInfoDatails Component 1`] = `
<DocumentFragment>
  <section
    class="ItemInfoDetails__container"
  >
    <h3
      class="ItemInfoDetails__title"
    >
      Liquid 1
    </h3>
    <p
      class="ItemInfoDetails__attribute"
    >
      <strong>
        Brand: 
      </strong>
       Acer
    </p>
    <p
      class="ItemInfoDetails__attribute"
    >
      <strong>
        SO: 
      </strong>
      windows
    </p>
    <p
      class="ItemInfoDetails__attribute"
    >
      <strong>
        Weight: 
      </strong>
      2kg
    </p>
    <p
      class="ItemInfoDetails__price"
    >
      Price: 
      <strong>
        16 â‚¬
      </strong>
    </p>
  </section>
</DocumentFragment>
`;
